### ISSUE TITLE
search should be case-insensitive

### ISSUE THREAD

**Issue by plughiubra** [fedora-infra/bodhi#997]
searching for "wxgtk3", for example, produces no results. the search should be case-insensitive, in line with user expectations about most search results.

**Comment by bowlofeggs** [comment]
I agree, thanks for the report!

**Comment by cosmologist10** [comment]
Hey! I want to work on this issue. Can you assist me?  
Thanks!

**Comment by bowlofeggs** [comment]
@cosmologist10 I'm not too familiar with this area of the code myself, but I think [this code](https://github.com/fedora-infra/bodhi/blob/2.2.4/bodhi/server/services/updates.py#L214-L217) is probably the issue:

```
    packages = data.get('packages')
    if packages is not None:
        query = query.join(Update.builds).join(Build.package)
        query = query.filter(or_(*[Package.name==pkg for pkg in packages]))
```

My guess is that `name==pkg` is probably the issue, and it should probably be changed to be a case insensitive match instead of a pure equality match.

**Comment by cosmologist10** [comment]
Thanks @bowlofeggs , I am working on this issue.

**Comment by cosmologist10** [comment]
Hey @bowlofeggs , There are two ways to work on  this issue :
1.  name.lower()==pkg.lower()
2.  import re
     re.match("pattern", "string", flags=re.IGNORECASE)

Can you suggest me which one is better, according to you.

**Comment by pnemade** [comment]
@cosmologist10 you need to give meaningful information for a commit log message. I see you just said "Solved" but that is not enough. The reviewer of your commit should get understanding by just reading the log message that for what purpose that commit has been added to the bodhi source.

For this fix your commit message should be saying "Change package search to case-insensitive"

**Comment by bowlofeggs** [comment]
@cosmologist10 Another option that would work well and that would improve the search otherwise is to use the ilike operator (i for case Insensitive, and otherwise the SQL like operator):

http://docs.sqlalchemy.org/en/latest/orm/internals.html#sqlalchemy.orm.properties.ColumnProperty.Comparator.ilike

**Comment by cosmologist10** [comment]
Thanks @bowlofeggs and @pnemade for the suggestions. I'm working on it.

**Comment by lalitkr** [comment]
Is this issue still open. I would like to work on it.

**Comment by bowlofeggs** [comment]
Hello @lalitkr Hmmm, I'm not sure. It seems that the pull request for this issue was closed. I do not recall merging a fix for this. @cosmologist10, do you still plan to work on this?

**Comment by cosmologist10** [comment]
Hello! I'm not working on this issue, You can assign it to @lalitkr.

**Comment by bowlofeggs** [comment]
While looking at #997 I decided to play around a bit to find what is happening. I realized that my earlier comment was not quite right in that there is more than one place in the code that we will need to fix this. I think we will need to amend a few more APIs by adding an "ilike" GET attr. When watching requests during a search on the development environment for "wxgtk", I saw four search requests get performed by the JS in the search box at the top of the WebUI:

```
/overrides/?like=wxgtk
/packages/?like=wxgtk
/updates/?like=wxgtk
/users/?like=wxgtk
```

For each of these, I think we will want to add a new ```ilike``` query that works very similarly to the existing "like" query, but uses the ilike database search instead of ```like```. We will need to keep the ```like``` query in place, since removing it would be backwards incompatible.

Once we have an ```ilike``` query, we will need to adjust the JS so that it uses the ```ilike``` query parameter instead. Additionally, we should take a look at the CLI so that it can use these new search types as well. It has the same problem:

```
$ bodhi updates query --packages wxgtk
0 updates found (0 shown)
$ bodhi updates query --packages wxGTK
wxGTK3-3.0.2-23.fc25                         bugfix       stable    2016-09-27
<snip>
```

Typing "wxgtk" into the search box and hitting enter queries ```/updates/?packages=wxgtk```, which was the API my previous comment was about. I think we don't want to change this query as I suggested in my previous comment, as that would be a backwards incompatible change.

**Comment by bowlofeggs** [comment]
One more thought - we could make hitting enter on the search box work with case insensitivity too by adding a new ```packages_ilike``` query parameter to the ```/updates/``` service, similar to the ilike I proposed. This way the old behavior can continue to be used by non-human actors that want the precision of specifying exact package names, but we can also give a nice human friendly search.

I had also been contemplating making it so that hitting enter in the search bar would still search users, overrides, and updates. We should do that in a separate effort though.

**Comment by bowlofeggs** [comment]
The patch for this issue should be included in the upcoming 2.10.0 release:

https://github.com/fedora-infra/bodhi/pull/1744

**Comment by keszybz** [comment]
This appears to have regressed. I'll open a new issue.
